tests/core/test_accounts.py

```diff
@@ -322,6 +322,63 @@ def test_eth_account_sign(acct, message, key, expected_bytes, expected_hash, v,
     assert account.signHash(msghash) == signed
 
 
+@pytest.mark.parametrize(
+    'message, message_type',
+    (
+        ("hello world", "string"),
+        (b'hello world', "bytes"),
+        ("68656c6c6f20776f726c64", "hexstr"),
+    )
+)
+def test_eth_account_sign_data_with_intended_validator(message, message_type, acct):
+    account = acct.create()
+
+    if message_type == "string":
+        primitive = None
+        text = message
+        hexstr = None
+    elif message_type == "bytes":
+        primitive = message
+        text = None
+        hexstr = None
+    elif message_type == "hexstr":
+        primitive = None
+        text = None
+        hexstr = message
+
+    with pytest.raises(TypeError):
+        # Raise TypeError if the address is more than 20 bytes
+        hashed_msg = defunct_hash_message(
```

Comment:
Also not a big deal, but you don't have to assign `hashed_msg` here. You can just say `defunct_hash_message(...`.